model/user/db.go

```diff
@@ -79,6 +85,34 @@ func FindOne(query db.Q) (*DBUser, error) {
 	return u, err
 }
 
+// FindOneByToken gets a DBUser by cached login token.
+func FindOneByToken(token string) (*DBUser, error) {
+	u := &DBUser{}
+	query := db.Query(bson.M{bsonutil.GetDottedKeyName(LoginCacheKey, LoginCacheTokenKey): token})
+	err := db.FindOneQ(Collection, query, u)
+	if err == mgo.ErrNotFound {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, errors.Wrap(err, "problem finding user by token")
+	}
+	return u, nil
+}
+
+// FindOneById gets a DBUser by ID.
+func FindOneById(id string) (*DBUser, error) {
+	u := &DBUser{}
+	query := ById(id)
+	err := db.FindOneQ(Collection, query, u)
+	if err == mgo.ErrNotFound {
+		return nil, nil
+	}
```

Comment:
do we have to?

Reply:
We don't. The distinction is meaningful for `FindOneByToken` though, so I did it here for symmetry. I'd prefer that `FindOneById` and `FindOneByToken` have similar behavior.